src/Features/LanguageServer/Protocol/Handler/Symbols/DocumentSymbolsHandler.cs

```diff
@@ -49,54 +46,65 @@ public override async Task<object[]> HandleRequestAsync(DocumentSymbolParams req
                 return Array.Empty<SymbolInformation>();
             }
 
-            var symbols = ArrayBuilder<object>.GetInstance();
-
-            var navBarService = document.Project.LanguageServices.GetRequiredService<Editor.INavigationBarItemService>();
-            var navBarItems = await navBarService.GetItemsAsync(document, cancellationToken).ConfigureAwait(false);
-            if (navBarItems.Count == 0)
+            var navBarService = document.Project.LanguageServices.GetRequiredService<INavigationBarItemService>();
+            var workspaceSupportsDocumentChanges = document.Project.Solution.Workspace.CanApplyChange(ApplyChangesKind.ChangeDocument);
+            var navBarItems = await navBarService.GetItemsAsync(document, workspaceSupportsDocumentChanges, cancellationToken).ConfigureAwait(false);
```

Comment:
fwiw, i would just return 'false' here.  unless you intend on supporting VB's use of nav bar to generate items (like event handlers).

Reply:
Yeah, I think you're probably right, especially since we won't get a callback for when the user actually goes to the location.